src/processors/BaseProcessor.cpp

```diff
@@ -241,3 +241,29 @@ juce::Point<int> BaseProcessor::getPosition (Rectangle<int> parentBounds)
 {
     return (editorPosition * juce::Point { (float) parentBounds.getWidth(), (float) parentBounds.getHeight() }).toInt();
 }
+
+bool BaseProcessor::hasModulationPorts()
+{
+    return modulationPorts;
+}
+
+void BaseProcessor::routeExternalModulation (int inputPort, int outputPort)
+{
+    jassert (numInputs >= 2 && numOutputs >= 2);
+    // If we're making modulation ports make sure there is actually a port for them
+
+    inputModulationPort = inputPort;
+    outputModulationPort = outputPort;
```

Comment:
Yeah I was thinking of something along these lines earlier as well, and was thinking of some very similar code with a container to hold the ports as well.

I couldn't think of any situations where having multiple modulation ports would be useful, but I would be curious if you have any in mind because there are probably some cool use cases I haven't thought of.

Reply:
I was thinking specifically about the chorus module, since it uses a 3-phase LFO... like we could have each phase go to a separate output if we want. That might be a bit extravagant for the chorus, but it would definitely be cool to have the option available if we want to do something like that for future modules.

Reply:
Ah smart, that would be really cool! Yeah it's not that much work to make it multi modulation input friendly.